### PR TITLE
Fix avatars in Notifications

### PR DESCRIPTION
**Pull request by iMattPro**
This should convert images that start with the relative root path into the board's absolute path.

Remains to be seen though if attached avatars (download/file.php) will work.

### CODE REVIEW

**Comment by rxu** [comment]
This will also require fixing `push_worker.js.twig` as far as currently
```
icon: response.avatar.src,
```

while now it should be just

```
icon: response.avatar,
```

**Comment by rxu** [comment]
Alternatively we could leave worker as is and just replace src in avatar img tag from relative to absolute path.

**Comment by rxu** [comment]
Also avatars in notifications are lazyloaded by default, so `phpbb_get_avatar()` returns `no_avatar.gif` in src, hence we will end up with no_avatar images for all avatars in webpushes.

**Comment by iMattPro** [comment]
That's why it's currently pulling the images from data-src first, then regular img src, thus bypassing the no avatar gifs

**Comment by rxu** [comment]
Got it. Then there's just `push_worker.js.twig` to fix.

File: notification/method/webpush.php [hunk 1]
```diff
@@ -428,4 +433,81 @@ protected function clean_expired_subscriptions(array $user_subscription_map, arr
...
+
+		// Extract filename using regex
+		$filename = '';
+		if (preg_match('/avatar=(.+?)_/', $file, $matches))
+		{
+			$filename = $matches[1];
+		}
+
+		// Construct final path
+		return $filename ? $this->path_helper->get_web_root_path() . $image_dir . DIRECTORY_SEPARATOR . $prefix . $filename . '.' . $ext : '';
```
Comment (rxu) [inline notification/method/webpush.php:496]: `DIRECTORY_SEPARATOR` is missing between `$this->path_helper->get_web_root_path()` and `$image_dir`. Also tested it on Windows and resulting path contains slashes and backslashes messed up, like
`http://localhost/33\images/avatars/upload\903172b946f8ce7e670825aa24ee8ed4_2.jpg`.

**Comment by rxu** [comment]
And there may be another problem: it seems that direct access to the files in `images/avatars/upload` is denied by `.htaccess` located within the directory. If so, this whole PR trick won't work.

**Comment by iMattPro** [comment]
@marc1706 We got all avatars working. How does this stuff look to you?

**Review by rxu — APPROVED** [review]

**Pull request merged**